fe/fe-core/src/main/java/org/apache/doris/scheduler/manager/AsyncJobManager.java

```diff
@@ -156,19 +156,20 @@ private void initAndSchedulerJob(Job job) {
     }
 
     private Long findFistExecuteTime(Long currentTimeMs, Long startTimeMs, Long intervalMs, boolean isCycleJob) {
+        // if job not delay, first execute time is start time
         if (startTimeMs != 0L && startTimeMs > currentTimeMs) {
             return startTimeMs;
         }
-        // if it's not cycle job and already delay, next execute time is current time
-        if (!isCycleJob) {
+        // if job already delay, first execute time is current time
+        if (startTimeMs != 0L && startTimeMs < currentTimeMs) {
             return currentTimeMs;
         }
```

Comment:
if starTime = 7, interval = 2, currentTime = 8:
user may hope the next execute time is 9 (7+2), rather than 8.

Reply:
good question. If the start time has been postponed, should we execute it in the next scheduling cycle, or immediately. If the cycle is very large, such as we schedule in weeks, then the next execution may be next week. Of course this will cause an overall delay in subsequent execution cycles.
This problem usually occurs when the machine is down and has not been restarted for a long time.

Reply:
ok